### ISSUE TITLE
Automatic Differentiation

### ISSUE THREAD

**Issue by rleegates** [JuliaDiff/HyperDualNumbers.jl#8]
Hi Rob,

I'd expect that hyper dual numbers could be extended to arbitrary order, or am I mistaken? Any plans on implementing this? I'd be willing to contribute methods for evaluating tensor derivatives :)

Best regards,
Robert

**Comment by mlubin** [comment]
You may want to look at https://github.com/jwmerrill/PowerSeries.jl and https://github.com/lbenet/TaylorSeries.jl.

**Comment by rleegates** [comment]
Thanks for the hint, Miles!

**Comment by goedman** [comment]
Hi Robert,

Apologies for not getting back sooner.

Yes, your are correct and extensions are possible and some examples are also shown on Jeff Fike's page at Stanford ( http://adl.stanford.edu/hyperdual/ ). 

At the moment I have no plans ( that means time! ) to implement these. I did this 'translation' as an initial project to work with Julia. 

But my goal has always been to implement Galerkin/Vianello like methods in Julia for buckling-of-pipes related issues and this will need to be my focus for the next year or so.

As Miles already pointed out, there are other options and I also really like ApproxFun.jl and SunDials.jl.

Regards,
Rob

**Comment by rleegates** [comment]
Hi Rob,

thanks for the input! Regarding the buckling of pipes, what kind of results are you interested in? Eigenvalues/vectors or postbuckling behavior? We have implemented a non-public finite element code for non-linear (stochastic) problems in Julia. With the implementation of an arc-length method, the extension of our contact algorithms to self-contact, and perhaps a suitable shell or beam element type, you'd be good to go :) 

Best,
Robert

**Comment by goedman** [comment]
Hi Robert,

Very interesting! How ‘non-public’ is your code? 

I’ve used eigenvalues/vectors (and Euler) for e.g. drill pipes in open salt domes to drill on to reach the next (deeper) salt dome.

But this work is what I call precursor-buckling (just how I have described it in some preliminary conversations). It has to do with pushing a pipe/casing in a long (horizontal) hole (up to 10 km to give a concrete example). This is on a ‘macro' scale, I believe related work deals with nano-tubes.

I have developed/used a Fortran/R based solution and while rewriting it in Julia I’m reviewing many of my basic assumptions and hope to verify at least partial results using e.g. Petr Krysl’s JFineale.jl and I.M. Smith & D. V. Griffith’s (Programming the finite element method) Fortran programs (a subset of which I’m also adapting to use within Julia for this purpose).

Regards,
Rob J. Goedman
goedman@mac.com
